### ISSUE TITLE
Focus is moved to the document body after selecting an image using the ImageInput component

### ISSUE THREAD

**Issue by robinmetral** [sumup-oss/circuit-ui#1705]
## Describe the bug

As title suggests. Discovered when testing https://github.com/sumup-oss/circuit-ui/pull/1644 (see https://github.com/sumup-oss/circuit-ui/pull/1644#discussion_r924490332)

## Steps to reproduce

1. Go to the [ImageInput story](https://circuit.sumup.com/iframe.html?id=forms-imageinput--base&viewMode=story) (as an iframe, easier for screen reader testing)
2. Turn on VoiceOver, upload an image
3. See error

## Expected behavior

The loading state should be announced and focus should later be moved back to the input element.

## Specifications

* Version: latest (5.2.3)
* Browser: Chrome
* Screen reader: VoiceOver
* OS: macOS

## Additional context

This might happen on other browser/SC combinations. Further investigation required.

**Comment by robinmetral** [comment]
The issue seems to be that the component is rerendering when an image is uploaded. That's not a huge problem on production since the browser is smart enough to pick up tabbing order from the place where focus was lost, but it would be best to prevent this to keep the experience as close as possible to an `input type=file`.
